File: multiple languages/2279. Maximum Bags With Full Capacity of Rocks/main.go
func maximumBags(capacity []int, rocks []int, additionalRocks int) int {

	toAdd := make([]int, len(capacity))
	for i := range toAdd {
		toAdd[i] = capacity[i] - rocks[i]
	}

	sort.Ints(toAdd)
	for i := 0; i < len(toAdd); i++ {
		if toAdd[i] > additionalRocks || additionalRocks == 0 {
			// here we can't fill current bag then the result will be equal to the previous index + 1 or current i
			return i
		}

		additionalRocks -= toAdd[i]
	}

	//we can fill all bags
	return len(toAdd)

}

File: multiple languages/2279. Maximum Bags With Full Capacity of Rocks/main.ts
function maximumBags(capacity: number[], rocks: number[], additionalRocks: number): number {

    let diff : number[] = new Array();

    for ( let i = 0; i < capacity.length; i++){
        diff[i] = capacity[i] - rocks[i];
    }

    diff.sort( (a:number,b:number) => a-b );

    //console.log(diff);

    for ( let i = 0; i < diff.length; i++ ){

        if (diff[i] > additionalRocks || additionalRocks === 0){
            return i ;
        }
        additionalRocks = additionalRocks - diff[i];
    }

    return diff.length;
};